# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/StoreBuffer.cpp
# Commit: 982a80b202c3
# Full Hash: 982a80b202c370876dafc72ac31be7a12ebeae58
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2021-01-08 03:45:03
# Regressor Bug: 1681533
# File Overlap Count: 2
# Description:
#   Bug 1681533 - Only collect the nursery when necessary in major GC r=sfink
#   
#   This rearranges a few things so that we can run a minor GC inside a major GC
#   slice, so we don't have to be conservative about collecting the nursery at the
#   start of a slice.
# ==============================================================================

diff -r 83f3726c41f9 -r 982a80b202c3 js/src/gc/StoreBuffer.cpp
--- a/js/src/gc/StoreBuffer.cpp	Thu Jan 07 15:57:41 2021 +0000
+++ b/js/src/gc/StoreBuffer.cpp	Mon Jan 04 17:01:50 2021 +0000
@@ -89,14 +89,13 @@
 {
 }
 
-void StoreBuffer::checkEmpty() const {
-  MOZ_ASSERT(bufferVal.isEmpty());
-  MOZ_ASSERT(bufStrCell.isEmpty());
-  MOZ_ASSERT(bufBigIntCell.isEmpty());
-  MOZ_ASSERT(bufObjCell.isEmpty());
-  MOZ_ASSERT(bufferSlot.isEmpty());
-  MOZ_ASSERT(bufferWholeCell.isEmpty());
-  MOZ_ASSERT(bufferGeneric.isEmpty());
+void StoreBuffer::checkEmpty() const { MOZ_ASSERT(isEmpty()); }
+
+bool StoreBuffer::isEmpty() const {
+  return bufferVal.isEmpty() && bufStrCell.isEmpty() &&
+         bufBigIntCell.isEmpty() && bufObjCell.isEmpty() &&
+         bufferSlot.isEmpty() && bufferWholeCell.isEmpty() &&
+         bufferGeneric.isEmpty();
 }
 
 bool StoreBuffer::enable() {